pandas/tests/frame/test_to_csv.py

```diff
@@ -1143,3 +1143,34 @@ def test_to_csv_quoting(self):
         df = df.set_index(['a', 'b'])
         expected = '"a","b","c"\n"1","3","5"\n"2","4","6"\n'
         self.assertEqual(df.to_csv(quoting=csv.QUOTE_ALL), expected)
+
+    def test_period_index_date_overflow(self):
+        # see gh-15982
+
+        dates = ["1990-01-01", "2000-01-01", "3005-01-01"]
+        index = pd.PeriodIndex(dates, freq="D")
+        df = pd.DataFrame([4, 5, 6], index=index)
+
+        buf = StringIO()
+        df.to_csv(buf)
+
+        expected = ',0\n1990-01-01,4\n2000-01-01,5\n3005-01-01,6\n'
+        self.assertEqual(buf.getvalue(), expected)
+
+        buf = StringIO()
+        date_format = "%m-%d-%Y"
+        df.to_csv(buf, date_format=date_format)
+
+        expected = ',0\n01-01-1990,4\n01-01-2000,5\n01-01-3005,6\n'
+        self.assertEqual(buf.getvalue(), expected)
+
+        # Overflow with pd.NaT
+        dates = ["1990-01-01", pd.NaT, "3005-01-01"]
+        index = pd.PeriodIndex(dates, freq="D")
+        df = pd.DataFrame([4, 5, 6], index=index)
+
+        buf = StringIO()
+        df.to_csv(buf)
```

Comment:
if you are looking for something to clean :>

there is a bit of a dichotomy between this test file and ``pandas/formats/test_formats.py`` where this tests 'higher' level functionaility and ``test_formats`` somewhat lower, but this is probably not consistent, and certainly not documented (maybe just a comment at the top of the file would be enough).